### ISSUE TITLE
2.4.0: third_party/QNNPACK is expected by the build but there is no such directry or git submodule

### ISSUE THREAD

**Issue by yurivict** [pytorch/pytorch#131864]
### 🐛 Describe the bug

The build fails:
```
Could not find any of CMakeLists.txt, Makefile, setup.py, LICENSE, LICENSE.md, LICENSE.txt in /usr/ports/misc/py-pytorch/work-py311/pytorch-v2.4.0/third_party/QNNPACK
Did you run 'git submodule update --init --recursive'?
```

Building with -DUSE_PYTORCH_QNNPACK=OFF doen't help.

### Versions

2.4.0

cc @malfet @seemethere @ZainRizvi @kit1980 @huydhn @clee2000

**Comment by janeyx99** [comment]
If you're building from source, could you clean your build first and retry?

e.g.,
```
python setup.py clean
python setup.py develop
```

Or if that doesn't work, recloning the repo should be worth a try.

**Comment by Geremia** [comment]
I get this issue, too, building/packaging with the source file: [pytorch-v2.4.0.tar.gz](https://github.com/pytorch/pytorch/releases/download/v2.4.0/pytorch-v2.4.0.tar.gz)
I can't run `git submodule update --init --recursive` because [pytorch-v2.4.0.tar.gz](https://github.com/pytorch/pytorch/releases/download/v2.4.0/pytorch-v2.4.0.tar.gz) is not a `git` repo.

**Comment by Geremia** [comment]
@yurivict 

> -DUSE_PYTORCH_QNNPACK=OFF doen't help

It seems to be hard-coded into the `setup.py`. I deleted the line: `sed -i '/"QNNPACK"/d' setup.py`, which helped.

**Comment by huydhn** [comment]
While 3rd-party QNNPACK was removed by https://github.com/pytorch/pytorch/pull/126657 a while ago, its reference in setup.py was cleaned up by https://github.com/pytorch/pytorch/pull/133177 only recently.

For release, let me cherry pick https://github.com/pytorch/pytorch/pull/133177 for the upcoming 2.4.1 release.

**Comment by malfet** [comment]
I'm curious how something like that got missed by CI, as we should validate that source package is buildable...

**Comment by kit1980** [comment]
This should be fixed already on 2.4.1
